### PR TITLE
Add ts templates for react

### PR DESCRIPTION
**Pull request by Krijovnick**

### CODE REVIEW

File: packages/devextreme-cli/src/templates/react/application/src/components/create-account-form/create-account-form.js [hunk 1]
```diff
@@ -28,11 +28,11 @@ export default function CreateAccountForm(props) {
     setLoading(false);
 
     if (result.isOk) {
-      history.push('/login');
+      navigate.push('/login');
```
[the remark this answers is not in the document]
Comment (Krijovnick) [reply packages/devextreme-cli/src/templates/react/application/src/components/create-account-form/create-account-form.js]: Fixed, I forget to remove the `push`

File: packages/devextreme-cli/src/templates/react/application/src/Content.js [hunk 2]
```diff
@@ -1,24 +1,25 @@
-import React from 'react';
-import { Switch, Route<%=^empty%>, Redirect<%=/empty%> } from 'react-router-dom';
+import { Routes, Route, Navigate } from 'react-router-dom';
 import appInfo from './app-info';
 import routes from './app-routes';
-import { <%=layout%> as SideNavBarLayout } from './layouts';
+import { SideNavOuterToolbar as SideNavBarLayout } from './layouts';
```
[the remark this answers is not in the document]
Comment (Krijovnick) [reply packages/devextreme-cli/src/templates/react/application/src/Content.js:4]: I will fix this in the next pr, where I use the `npm run update-template` command to generate template for cli that generate this variable.

File: packages/devextreme-cli/src/templates-ts/react/application/src/contexts/auth.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+type IAuthContextType = {
+  user?: IUser;
+  signIn: (email: string, password: string) => Promise<{isOk: boolean, data?: IUser, message?: string}>;
+  signOut: () => void;
+  loading: boolean;
+}
+
+interface IAuthProviderProps {
```
Comment (MikeVitik) [inline packages/devextreme-cli/src/templates-ts/react/application/src/contexts/auth.tsx:16]: Don't add new type children: React.ReactNode; use existing

File: packages/devextreme-cli/src/templates-ts/react/application/src/contexts/auth.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      const result = await getUser();
+      if (result.isOk) {
+        setUser(result.data);
+      }
+
+      setLoading(false);
+    })();
+  }, []);
+
+  const signIn = useCallback(async (email: string, password: string) => {
```
Comment (MikeVitik) [inline packages/devextreme-cli/src/templates-ts/react/application/src/contexts/auth.tsx:35]: If setUser doesn't mutate we should the useCallback if mutate we should add it to the dependancies

**Review by MikeVitik — CHANGES_REQUESTED** [review]

File: packages/devextreme-cli/src/templates-ts/react/application/src/app-navigation.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
+export const navigation = [<%=^empty%>
```
Comment (dxvladislavvolkov) [inline packages/devextreme-cli/src/templates-ts/react/application/src/app-navigation.tsx:1]: Maybe we need to add a navigation type?

File: packages/devextreme-cli/src/templates-ts/react/application/src/app-routes.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { HomePage, TasksPage, ProfilePage } from './pages';
+import { withNavigationWatcher } from './contexts/navigation';
+
+const routes = [
```
Comment (dxvladislavvolkov) [inline packages/devextreme-cli/src/templates-ts/react/application/src/app-routes.tsx:4]: The same for routes. Maybe add a route type?

File: packages/devextreme-cli/src/templates-ts/react/application/src/components/header/header.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+          >
+            <UserPanel menuMode={'context'} />
+          </Button>
+        </Item>
+        <Template name={'userPanelTemplate'}>
+          <UserPanel menuMode={'list'} />
+        </Template>
+      </Toolbar>
+    </header>
+)}
```
Comment (dxvladislavvolkov) [inline packages/devextreme-cli/src/templates-ts/react/application/src/components/header/header.tsx:51]: add a new line

File: packages/devextreme-cli/src/templates/react/application/src/Content.js [hunk 8]
```diff
@@ -1,24 +1,25 @@
...
+      <Routes>
+        {routes.map(({ path, element }) => (
           <Route
-            exact
             key={path}
             path={path}
-            component={component}
+            element={element}
           />
-        ))}<%=^empty%>
```
Comment (dxvladislavvolkov) [inline packages/devextreme-cli/src/templates/react/application/src/Content.js:19]: Will this also be fix in the next pr?

File: packages/devextreme-cli/src/templates/react/application/src/app-routes.js [hunk 9]
```diff
@@ -1,24 +1,24 @@
...
-  },
-  {
-    path: '/profile',
-    component: ProfilePage
-  },
-  {
-    path: '/home',
-    component: HomePage
-  }
-<%=/empty%>];
```
Comment (dxvladislavvolkov) [inline packages/devextreme-cli/src/templates/react/application/src/app-routes.js:17]: and this?

File: packages/devextreme-cli/src/templates-ts/react/application/src/components/header/header.tsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import Toolbar, { Item } from 'devextreme-react/toolbar';
+import Button from 'devextreme-react/button';
+import UserPanel from '../user-panel/user-panel';
+import './header.scss';
+import { Template } from 'devextreme-react/core/template';
+
+interface IHeaderProps {
```
[the remark this answers is not in the document]
Comment (Krijovnick) [reply packages/devextreme-cli/src/templates-ts/react/application/src/components/header/header.tsx:8]: Ok, I remove it in the next pr

File: packages/devextreme-cli/src/templates-ts/react/application/src/components/reset-password-form/reset-password-form.tsx [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+import './reset-password-form.scss';
+
+const notificationText = 'We\'ve sent a link to reset your password. Check your inbox.';
+
+export default function ResetPasswordForm() {
+  const navigate = useNavigate();
+  const [loading, setLoading] = useState(false);
+  const formData = useRef({ email: '', password: '' });
+
+  const onSubmit = useCallback(async (e) => {
```
[the remark this answers is not in the document]
Comment (Krijovnick) [reply packages/devextreme-cli/src/templates-ts/react/application/src/components/reset-password-form/reset-password-form.tsx:23]: It isn't my code, it is copy past. I will research why it was made like this in the next pr

File: packages/devextreme-cli/src/templates-ts/react/application/src/components/side-navigation-menu/side-navigation-menu.tsx [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+    openMenu,
+    compactMode,
+    onMenuReady
+  } = props;
+
+  const { isLarge } = useScreenSize();
+  function normalizePath () {
+    return navigation.map((item) => {
+      if(item.path && !(/^\//.test(item.path))){
+        item.path = `/${item.path}`;
```
[the remark this answers is not in the document]
Comment (Krijovnick) [reply packages/devextreme-cli/src/templates-ts/react/application/src/components/side-navigation-menu/side-navigation-menu.tsx:31]: It isn't my code, it is copy past. I will fix it in the next pr

File: packages/devextreme-cli/src/templates-ts/react/application/src/components/side-navigation-menu/side-navigation-menu.tsx [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+      }
+      return {...item, expanded: isLarge};
+    });
+  }
+
+  const items = useMemo(
```
[the remark this answers is not in the document]
Comment (Krijovnick) [reply packages/devextreme-cli/src/templates-ts/react/application/src/components/side-navigation-menu/side-navigation-menu.tsx:37]: It isn't my code, it is copy past. I will research why it was made like this in the next pr

File: packages/devextreme-cli/src/templates-ts/react/application/src/contexts/auth.tsx (hunk 4 above)
Comment (Krijovnick) [reply packages/devextreme-cli/src/templates-ts/react/application/src/contexts/auth.tsx:35]: It isn't my code, it is copy past. I will research why it was made like this in the next pr

File: packages/devextreme-cli/src/templates-ts/react/application/src/app-navigation.tsx (hunk 5 above)
Comment (Krijovnick) [reply packages/devextreme-cli/src/templates-ts/react/application/src/app-navigation.tsx:1]: The typescript does not underline with an error the navigation. Should we add type where typescript understand this type?

File: packages/devextreme-cli/src/templates-ts/react/application/src/contexts/auth.tsx (hunk 3 above)
Comment (Krijovnick) [reply packages/devextreme-cli/src/templates-ts/react/application/src/contexts/auth.tsx:16]: Ok, I will fix it in the next pr

File: packages/devextreme-cli/src/templates-ts/react/application/src/app-routes.tsx (hunk 6 above)
Comment (Krijovnick) [reply packages/devextreme-cli/src/templates-ts/react/application/src/app-routes.tsx:4]: The same. Typescript understand this type.

File: packages/devextreme-cli/src/templates-ts/react/application/src/components/header/header.tsx (hunk 7 above)
Comment (Krijovnick) [reply packages/devextreme-cli/src/templates-ts/react/application/src/components/header/header.tsx:51]: Ok, I will fix it in the next pr

File: packages/devextreme-cli/src/templates/react/application/src/Content.js (hunk 8 above)
Comment (Krijovnick) [reply packages/devextreme-cli/src/templates/react/application/src/Content.js:19]: Yes

File: packages/devextreme-cli/src/templates/react/application/src/app-routes.js (hunk 9 above)
Comment (Krijovnick) [reply packages/devextreme-cli/src/templates/react/application/src/app-routes.js:17]: Yes

**Review by MikeVitik — APPROVED** [review]

**Pull request merged**